main.tf

```diff
@@ -0,0 +1,101 @@
+terraform {
+    required_providers {
+        azurerm = {
+          source = "hashicorp/azurerm"
+          version = ">= 3.8"
+        }
+    }
+    backend "azurerm" {
+        resource_group_name  = "Cohort22_JonLon_ProjectExercise"
+        storage_account_name = "tfstate1147207684"
+        container_name       = "tfstate"
+        key                  = "$env:ARM_ACCESS_KEY"
```

Comment:
Really unhelpful naming here, but the `key` actually just refers to a filename - it's a "key" in the same way a dictionary has "key-value" pairs, rather than referring to a secret. If you look in Azure, you'll find that in your storage account there's literally a file called `$env:ARM_ACCESS_KEY` - so no real harm but this isn't actually doing what you think!

Reply:
renamed key to FileName

Reply:
Doh this is required to be Key but is just a filename